tests/harness/model/config.py

```diff
@@ -0,0 +1,129 @@
+    """ZenML configuration settings."""
+
+    deployments: List[DeploymentConfig] = Field(default_factory=list)
+    secrets: List[Secret] = Field(default_factory=list)
+    tests: List[TestConfig] = Field(default_factory=list)
+    requirements: List[TestRequirements] = Field(default_factory=list)
+    environments: List[EnvironmentConfig] = Field(default_factory=list)
+
+    _config_file: str
+
+    def __init__(self, config_file: str, **data: Any) -> None:
+        self._config_file = config_file
+        super().__init__(**data)
+
+    def merge(self, config: "Configuration") -> None:
+        """Updates the configuration with the contents of another configuration.
+
+        Args:
+            config: The configuration to merge into this one.
+        """
+
+        def check_duplicate_keys(
+            entries: Sequence[BaseModel], key_attr: str
+        ) -> List[str]:
+            """Checks for duplicated keys.
+
+            Args:
+                entries: List of pydantic objects.
+                key_attr: The attribute to use as key.
+
+            Returns:
+                A list of duplicate keys.
+            """
+            keys = [getattr(entry, key_attr) for entry in entries]
+            return [key for key in keys if keys.count(key) > 1]
+
+        self.deployments += config.deployments
+        self.secrets += config.secrets
+        self.tests += config.tests
+        self.requirements += config.requirements
+        self.environments += config.environments
+
+        duplicates = check_duplicate_keys(self.deployments, "name")
+        if duplicates:
+            raise ValueError(
+                f"Configuration error: deployments with duplicate "
+                f"names loaded from configuration file `{self._config_file}`: "
+                f"{', '.join(duplicates)}"
+            )
+        duplicates = check_duplicate_keys(self.secrets, "name")
+        if duplicates:
+            raise ValueError(
+                f"Configuration error: secrets with duplicate "
+                f"names loaded from configuration file `{self._config_file}`: "
+                f"{', '.join(duplicates)}"
+            )
+        duplicates = check_duplicate_keys(self.tests, "module")
+        if duplicates:
+            raise ValueError(
+                f"Configuration error: test requirements entries with "
+                f"duplicate module names loaded from configuration file "
+                f"`{self._config_file}`: {', '.join(duplicates)}"
+            )
+        duplicates = check_duplicate_keys(self.requirements, "name")
+        if duplicates:
+            raise ValueError(
+                f"Configuration error: global requirements entries with "
+                f"duplicate names loaded from configuration file "
+                f"`{self._config_file}`: {', '.join(duplicates)}"
+            )
+        duplicates = check_duplicate_keys(self.environments, "name")
+        if duplicates:
```

Comment:
this looks like it could be done a bit cleaner in a for loop

Reply:
You're right. I typed a `for item_list ` at the top of this and Github copilot wrote the whole compressed for loop for me. Crazy scary !

Reply:
You're right. I typed a `for item_list ` at the top of this and Github copilot wrote the whole compressed for loop for me. Crazy scary !